batchlib/mongo/result_writer.py

```diff
@@ -0,0 +1,137 @@
+    # document attributes cannot contain '.'
+    column_names = [cn.replace('.', '_') for cn in column_names]
+    return [
+        dict(zip(column_names, table_row)) for table_row in table
+    ]
+
+
+def _get_result_tables(in_file):
+    with h5py.File(in_file, 'r') as f:
+        result_tables = {}
+        for table_name in _get_table_names(f):
+            column_names, table = read_table(f, table_name)
+            result_tables[table_name] = _table_object(column_names, table)
+
+        return result_tables
+
+
+def _get_log_path(work_dir):
+    logs = list(glob.glob(os.path.join(work_dir, '*.log')))
+    assert len(logs) == 1
+    return logs[0]
+
+
+def _parse_workflow_duration(work_dir):
+    """
+    Reads workflow duration by parsing the first and the last log event in the log file and taking the time difference
+    """
+    log_path = _get_log_path(work_dir)
+    with open(log_path, 'r') as fh:
+        lines = list(fh)
+        for first_log in lines:
+            if 'INFO' in first_log:
+                break
+
+        for last_log in reversed(lines):
+            if 'INFO' in last_log:
+                break
+
+        start = datetime.strptime(first_log.split('[')[0].strip(), '%Y-%m-%d %H:%M:%S')
+        end = datetime.strptime(last_log.split('[')[0].strip(), '%Y-%m-%d %H:%M:%S')
+
+        delta = end - start
+        return delta.seconds
+
+
+def _parse_workflow_name(work_dir):
+    """
+    Parses workflow name from the log file
+    """
+    log_path = _get_log_path(work_dir)
+    logfile = os.path.split(log_path)[1]
+    workflow_name = os.path.splitext(logfile)[0]
+    return workflow_name
+
+
+def _get_git_sha():
+    try:
+        repo = git.Repo(search_parent_directories=True)
+        return repo.head.object.hexsha
+    except Exception as e:
+        logger.warning(f'Cannot get batchlib_version: {e}')
+        return None
+
+
+class DbResultWriter(BatchJobOnContainer):
+    def __init__(self, username, password, host, port=27017, db_name='covid', **super_kwargs):
+        super().__init__(input_pattern='*.hdf5', **super_kwargs)
+
+        username = urllib.parse.quote_plus(username)
+        password = urllib.parse.quote_plus(password)
+
+        mongodb_uri = f'mongodb://{username}:{password}@{host}:{port}/?authSource={db_name}'
+
+        logger.info(f'Connecting to MongoDB instance: {host}:{port}, user: {username}, authSource: {db_name}')
+
+        try:
+            client = MongoClient(mongodb_uri, serverSelectionTimeoutMS=5000)
+            logger.info(f'Server info: {client.server_info()}')
+            self.db = client[db_name]
+        except Exception as e:
+            logger.warning(f'Connection failure: {e}. Skipping DbResultWriter job.')
+            self.db = None
+
+    def run(self, input_files, output_files, **kwargs):
+        if self.db is None:
+            return
+
+        assert len(input_files) == 1, f'Expected a single table file, but {len(input_files)} was given'
```

Comment:
I think you brought this up somewhere else already, but we might have multible `..._tables.hdf5` now.
We will also need to dicsuss with @imagirom and @tichi how to handle this.

Reply:
that's not a problem: I can just have another level below `result_tables` object and write all the tables in there. Schema would be:
```
...
result_tables: {
   table_name1: {
      'wells/default': {...},
      ...
   },
   table_name2: {
   }
   ...
}
```
I'll do that now. How to handle this in the plate viewer is a different issue, that we'll discuss in https://github.com/hci-unihd/batchlib/issues/84

Reply:
Ok this is done, the job supports multiple analysis tables (hdf5 files) now. The schema is adapted as well and described in the README. Sample output from the DB:
```
> coll.findOne()
{
	"_id" : ObjectId("5eaec67bccb0dff4c1f1b3a6"),
	"created_at" : ISODate("2020-05-03T15:26:19.988Z"),
	"workflow_name" : "CellAnalysisWorkflow",
	"workflow_duration" : 86,
	"plate_name" : "test-cell-analysis",
	"batchlib_version" : "7b4eb40891e5b3b84737776a9750e1a3d6c38ac6",
	"result_tables" : [
		{
			"analysis_name" : "test-cell-analysis_table_serum_corrected",
			"tables" : [
				{
					"table_name" : "images/default",
					"results" : [
						{
...
}
```

Reply:
:white_check_mark: